fastlane/lanes/localization.rb

```diff
@@ -122,49 +127,78 @@
   # @called_by complete_code_freeze
   #
   lane :generate_strings_file_for_glotpress do |options|
-    cocoapods
-
-    # On top of fetching the latest Pods, we also need to fetch the source for the Gutenberg code.
-    # To get it, we need to manually clone the repo, since Gutenberg is distributed via XCFramework.
-    # XCFrameworks are binary targets and cannot extract strings via genstrings from there.
-    config = gutenberg_config!
-
-    ref_node = config[:ref]
-    UI.user_error!('Could not find Gutenberg ref to clone the repository in order to access its strings.') if ref_node.nil?
-
-    ref = ref_node[:tag] || ref_node[:commit]
-    UI.user_error!('The ref to clone Gutenberg in order to access its strings has neither tag nor commit values.') if ref.nil?
-
-    github_org = config[:github_org]
-    UI.user_error!('Could not find GitHub organization name to clone Gutenberg in order to access its strings.') if github_org.nil?
-
-    repo_name = config[:repo_name]
-    UI.user_error!('Could not find GitHub repository name to clone Gutenberg in order to access its strings.') if repo_name.nil?
-
-    # Create a temporary directory to clone Gutenberg into.
-    # We'll run the rest of the automation from within the block, but notice that only the Gutenbreg cloning happens within the temporary directory.
-    gutenberg_clone_name = 'Gutenberg-Strings-Clone'
+    # We jump in the tempdir immediately, even though we might not need it.
+    # It's just simpler to rely on the Ruby block API for it than to manage it by hand.
     Dir.mktmpdir do |tempdir|
-      Dir.chdir(tempdir) do
-        repo_url = "https://github.com/#{github_org}/#{repo_name}"
-        UI.message("Cloning Gutenberg from #{repo_url} into #{gutenberg_clone_name}. This might take a few minutes…")
-        sh("git clone --depth 1 #{repo_url} #{gutenberg_clone_name}")
-        Dir.chdir(gutenberg_clone_name) do
-          if config[:ref][:tag]
-            sh("git fetch origin refs/tags/#{ref}:refs/tags/#{ref}")
-            sh("git checkout refs/tags/#{ref}")
-          else
-            sh("git fetch origin #{ref}")
-            sh("git checkout #{ref}")
+      # Fetch fresh pods to read the latest localizations from them.
+      # In CI, we expect the pods to be already available and up to date.
+      cocoapods unless is_ci
+
+      # For the same reason, fetch fresh packages
+      # FIXME: We yet don't have an explicit way to set the derived data in CI, so we'll run it in CI, too
```

Comment:
If/when we had that option, then I'd have something like 

```ruby
UI.user_error!("CI detected but no derived_data_path provided.") if is_ci && options[:derived_data_path].nil?
```

Then at the point where the CI script invokes this lane (or its caller),

```bash
DERIVED_DATA_PATH=./DerivedData
install_swift_dependencies "$DERIVED_DATA_PATH"

bundle exec fastlane ... "derived_data_path:$DERIVED_DATA_PATH"
```

Reply:
Oooooh `<epiphany>` I just realized what you meant after all.

The issue is that the `install_swiftpm_dependencies` ci-toolkit utility doesn't (yet) expose a way for us to provide a custom DerivedData where to install those dependencies to, so that means that when that was run from the `.sh` before calling fastlane the dependencies were installed in Xcode's default DerivedData while in the rest of this code invoked from Fastlane we might want to use a different DerivedData location `</epiphany>`

Ok that now makes more sense to me why you've re-implemented the logic again via `resolve_packages` lane below now, got it.

---

The good news a that SwiftPM always first create a bare mirror of the SPM repos (`git clone --bare …`) in `~/Library/Caches/org.swift.swiftpm/repositories` [^1] **regardless of the location specified by `xcodebuild` for DerivedData**; and _only then_ it does a proper checkout[^2] from that bare repo / local mirror repository into the specified DerivedData folder.

Which means that, as long as you use `-useVersionsFromResolvedFile`, the checkout will be super fast (almost instant), **even if the DerivedData path was not the same as when we called `install_swiftpm_dependencies`** to restore the SPM cache, because it will create a checkout[^2] of the files in the DerivedData folder… from the **local** bare repository[^1] in `~/Library/Caches`, as opposed to redownloading them from the net.

So TL;DR the SPM _cache_ and s independent of the DerivedData path (only the _checkout_ to transform the bare repo from the cache into a checkout into DerivedData is)

[^1]: A bare git repo is basically only the content of the `.git` folder you're used to, with all the repo history and git objects/refs/…, but without the human-browsable files they represent in the filesystem.

[^2] In this context, the term "git checkout" is used in the strict sense of the git terminology, and basically consists of git creating human-browsable files on the filesystem representing the state of the objects in the `.git/objects` database.

Reply:
Good point. Here's my understanding:

We don't yet have a way to ensure that `install_swift_dependencies` and `generate_strings_file_for_glotpress` use the same derived data folder from CI. That's because `install_swift_dependencies` doesn't accept a value to pass to `xcodebuild` via `-derivedDataPath` ( 🤓🏹 )

If we could, I presume we would be able to bypass calling `resolve_packages` if `is_ci` and pass the path to the derived data used by `install_swift_dependencies`, where we would fine the SPM sources.

Since that's not possible yet, we need to resolve the packages in CI within the lane, just to make sure.

Reply:
> The thing about xcodebuild using ~/Library/Caches was a TIL for me.

Yep, I learned this (+ a lot about git repositories vs checkouts) not so long ago while I was looking a bit more in details how SwiftPM caching logic worked while working on https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/105 (and because I was wondering if the DerivedData used on CI would affect the location we'd need to cache in our `install_swiftpm_dependencies` implementation)

I actually plan to write about these in a P2 soon.